app/src/main/java/com/alexchan/wallpaper/ui/search/searchResults/SearchResultsActivity.kt

```diff
@@ -0,0 +1,87 @@
+package com.alexchan.wallpaper.ui.search.searchResults
+
+import android.app.SearchManager
+import android.content.Intent
+import android.os.Bundle
+import android.provider.SearchRecentSuggestions
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.GravityCompat
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.navigation.fragment.findNavController
+import com.alexchan.wallpaper.R
+import com.alexchan.wallpaper.database.SearchSuggestionProvider
+import com.alexchan.wallpaper.ui.mainActivity.MainActivity
+import com.alexchan.wallpaper.ui.mainActivity.MainActivity.Companion.searchQuery
+import com.alexchan.wallpaper.ui.mainActivity.MainActivity.Companion.searchStatus
+import kotlinx.android.synthetic.main.activity_main.*
+
+class SearchResultsActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // Set Search Wallpaper Navigation Host
+        val navHostFragment = supportFragmentManager.findFragmentById(R.id.mainNavHostFragment)
+        val navController = navHostFragment?.findNavController()
+        navController?.setGraph(R.navigation.search_wallpaper)
```

Comment:
Not sure how it works with navigation component to navigate to other activity, but there's many code in this class can be simplified if using fragment. Let me know if there is any reason using activity.

Reply:
Since I'm reusing the dashboard to display search result and dashboard_details to display search result details, if I do not create a new activity, the search results will just display on the same screen as previously implemented. It also uses these meta data and intent filter to get data from the SearchManager. The above navController uses a new Graph, called search_wallpaper, which reuses the dashboard and dashboard_details to only handle user search query.

```
<intent-filter> 
<action android:name="android.intent.action.SEARCH" android:launchMode="singleTop" /> 
</intent-filter> 
<meta-data android:name="android.app.searchable" android:resource="@xml/searchable" /> 
<intent-filter> 
<action android:name="android.intent.action.VIEW" /> 
</intent-filter>
```